---
title: "pQTL <-> scz"
author: X Shen  
date: "\n`r format(Sys.time(), '%d %B, %Y')`"
output: github_document
always_allow_html: true
---

```{r library,echo=F,warning=F,error=F,message=F}
library(rmarkdown)
library(dplyr)
library(data.table)
library(pbapply)
library(readr)
library(Hmisc)
library(here)
library(knitr)
library(kableExtra)
library(TwoSampleMR)
#library(MungeSumstats)
#library(biomaRt)

```

## Prepare SCZ sumstats

```{r, read MDD gwas sumstats}
scz.gwas = read_tsv('data/daner_PGC_SCZ_w3_90_0518d_eur.gz')
```

```{r, rename MDD sumstats colnames}
scz.gwas = scz.gwas %>% 
 mutate(beta = log(OR)) %>% 
 dplyr::select(SNP,effect_allele=A1,other_allele=A2,
                       eaf=FRQ_U_77258,beta,se=SE,pval=P,samplesize=Neff) %>% 
         mutate(Phenotype='SCZ') 
```

## Analysis: pQTL -> SCZ

### Format SCZ (outcome)

```{r, format MDD sumstats as outcome}
dat.scz.outcome = scz.gwas %>% 
  format_data(.,type='outcome')
```

### Format pQTL data (exposure)

Download pQTL annotation file from [here](https://download.decode.is/form/2021/assocvariants.annotated.txt.gz)

```{r,format pQTL as exposure data and clump}
annot_variant = read_tsv('data/assocvariants.annotated.txt.gz') %>%
    dplyr::select(SNP=rsids,effect_allele=effectAllele,eaf=effectAlleleFreq)

format_pqtl_exposure <- function(f.x.dat,chr,pos0,pos1){
  Symbol = f.x.dat %>% gsub('_10032022.txt.gz','',.) %>% basename 
  x.dat = read_tsv(f.x.dat) %>%
    dplyr::filter(Chrom!='chrX') %>%
    mutate(CHR=Chrom %>% gsub('chr','',.) %>% as.numeric,
           p=10^(-minus_log10_pval)) %>%
    dplyr::filter(ImpMAF>0.001) %>% 
    dplyr::select(SNP=rsids,CHR,BP=Pos,MAF=ImpMAF,A1=effectAllele,A2=otherAllele,beta=Beta,se=SE,p,N) %>% 
    MungeSumstats::liftover(sumstats_dt = ., 
                            ref_genome = "hg38",
                            convert_ref_genome = "hg19",end_col='BP') %>% 
    mutate(Phenotype=Symbol) %>%
    dplyr::select(CHR,BP,SNP,effect_allele=A1,other_allele=A2,beta,se,pval=p,samplesize=N,Phenotype) %>% 
    .[.$SNP %in% scz.gwas$SNP,] %>%
    dplyr::filter(CHR==chr,BP>pos0-1000000,BP<pos1+1000000) %>% 
    left_join(.,annot_variant,by=c('SNP','effect_allele'))

    x.dat.clumped = x.dat %>% 
      filter(pval<5e-8) %>% 
      as.data.frame %>%
      format_data(.,type='exposure') %>% 
      clump_data(.,clump_kb=1000,clump_r2=0.001,pop='EUR') %>%
      arrange(pval.exposure) %>%
      .[1,]
      return(x.dat.clumped)
  
}

dat.mqtl.exposure = list.files(here::here('data/deCODE_pQTL/'),full.names = T) %>% 
  .[grep('.gz$',.)] %>%
  as.list %>% 
  pblapply(format_pqtl_exposure)
  ```

Prepare annotation file

Get Ensembl gene annotation, check tutorial [here](https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/accessing_ensembl.html)

```{r}
# Map to ensembl IDs
## Get reference ensembl database
mart_prot <- useMart("ensembl", dataset = "hsapiens_gene_ensembl") # human
Uniprot_ensembl = getBM(
  attributes=c('ensembl_gene_id','uniprotswissprot',"hgnc_symbol","start_position","end_position"), 
  mart = mart_prot)
colnames(Uniprot_ensembl) <- c("Ensembl_ID", "UniProt","Gene_symbol","start_pos","end_pos")


```