---
title: "pQTL, SCZ GWAS sumstats, reference data preparation"
author: X Shen  
date: "\n`r format(Sys.time(), '%d %B, %Y')`"
output: github_document
always_allow_html: true
---

```{r library,echo=F,warning=F,error=F,message=F}
# Note that if you want to run this script on your local machine, you may need to install additional packages listed down here.

library(rmarkdown)
library(dplyr)
library(data.table)
library(pbapply)
library(readr)
library(Hmisc)
library(here)
library(knitr)
library(kableExtra)
library(TwoSampleMR)
library(MungeSumstats)
library(biomaRt)
library(stringr)
```

# Prepare annotation file

## Files required for our analysis

-   Protein assay annotations. Key information include: UniProt ID, gene name, gene Ensembl IDs, SeqIDs. This is usually assay-specific. deCODE proteomic data used a SOMALogic assay. More information can be found in the [paper](https://www.nature.com/articles/s41588-021-00978-w). We will use their [Supplementary Table 1](https://static-content.springer.com/esm/art%3A10.1038%2Fs41588-021-00978-w/MediaObjects/41588_2021_978_MOESM4_ESM.xlsx).

    The protein annotation table looks this:

    | SeqId    | Protein (short name) | Protein (full name)                                | Gene   | UniProt | Organism | Type    | Ensembl.Gene.ID |
    | -------- | -------------------- | -------------------------------------------------- | ------ | ------- | -------- | ------- | --------------- |
    | 10000_28 | CRBB2                | Beta-crystallin B2                                 | CRYBB2 | P43320  | human    | Protein | ENSG00000244752 |
    | 10001_7  | c-Raf                | RAF proto-oncogene serine/threonine-protein kinase | RAF1   | P04049  | human    | Protein | ENSG00000132155 |
    | 10003_15 | ZNF41                | Zinc finger protein 41                             | ZNF41  | P51814  | human    | Protein | ENSG00000147124 |

    A local copy is stored in utils/ref_somalogic.tsv



-   Gene annotations. Key information is start and end genomic positions of a given protein-encoding gene. We can query using the 'bioMart' R package. Check the tutorial [here](https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/accessing_ensembl.html). View online for more info [here](https://www.ensembl.org/info/data/biomart/how_to_use_biomart.html). 

```{r, echo=T,warning=F,message=F}
# Map to ensembl IDs
## Get reference ensembl database
mart_prot <- useMart("ensembl", dataset = "hsapiens_gene_ensembl") # human
Uniprot_ensembl = getBM(
  attributes=c('ensembl_gene_id','uniprotswissprot',"hgnc_symbol","chromosome_name","start_position","end_position"), 
  mart = mart_prot)
colnames(Uniprot_ensembl) <- c("Ensembl_ID", "UniProt","Gene_symbol","CHR","start_pos","end_pos")

knitr::kable(Uniprot_ensembl %>% filter(UniProt!='') %>% arrange(CHR) %>% head(5), "pipe")

write_tsv(Uniprot_ensembl,here::here('utils/uniprot_ensembl.tsv'))
```

    A local copy of Ensembl annotation is stored in utils/uniprot_ensembl.tsv


-   Variants annotation for pQTL data

    Sometimes researchers choose to store additional variant information, such as allele frequency, that tend to be consistent across multiple analyses, separately from individual GWAS sumstats. 

    You can request this information for the pQTL data from [here](https://download.decode.is/form/2021/assocvariants.annotated.txt.gz).

    A copy of this information is stored on cloud.


# Prepare pQTL data

```{r,format pQTL,echo=T,eval=F}
annot_variant = read_tsv('data/assocvariants.annotated.txt.gz') %>%
    dplyr::select(eaf=effectAlleleFreq,Name) 

ref.somalogic = read_tsv(here::here('utils/ref_somalogic.tsv')) %>%
    left_join(.,read_tsv(here::here('utils/uniprot_ensembl.tsv')),by='UniProt') %>%
    mutate(CHR=as.numeric(CHR))

format_pqtl_exposure <- function(f.x.dat,x.ref){
  # Get annotation
  Symbol = f.x.dat %>% basename %>% gsub('_10032022.txt.gz','',.) %>% gsub('Proteomics_SMP_PC0_','',.) 
  x.seqid = Symbol %>% str_split(.,'_') %>% .[[1]] %>% head(2) %>% paste0(.,collapse='_')
  x.ref = x.ref %>% filter(SeqId == x.seqid)
  x.chr = x.ref$CHR
  x.start = x.ref$start_pos
  x.end = x.ref$end_pos

  # Output name
  f.out = f.x.dat %>% basename %>% gsub('.txt.gz','_lo_annot.txt.gz',.)

  # Liftover from hg38 to hg19
  x.dat.liftover = read_tsv(f.x.dat) %>%
    dplyr::filter(Chrom!='chrX') %>%
    mutate(CHR=Chrom %>% gsub('chr','',.) %>% as.numeric,
           p=10^(-minus_log10_pval)) %>%
    dplyr::select(SNP=rsids,CHR,BP=Pos,MAF=ImpMAF,A1=effectAllele,A2=otherAllele,beta=Beta,se=SE,p,N,Name) %>% 
    MungeSumstats::liftover(sumstats_dt = ., 
                            ref_genome = "hg38",
                            convert_ref_genome = "hg19",end_col='BP') %>% 
    mutate(Phenotype=Symbol) 

  # Add variant annotation
  x.dat.liftover_annot = x.dat.liftover %>%
    dplyr::select(CHR,BP,SNP,effect_allele=A1,other_allele=A2,beta,se,pval=p,samplesize=N,Phenotype,Name) %>% 
    dplyr::filter(CHR==x.chr) %>% 
    left_join(.,annot_variant,by=c('Name')) %>%
    dplyr::select(-Name)
  
  # Write preprocessed file
  dir.create(here::here('./data/pQTL_hg19_annot/'), showWarnings = FALSE)
  write_tsv(x.dat.liftover_annot,here::here(paste0('data/pQTL_hg19_annot/',f.out)))
}

# Prepare pQTL data and write files in data/pQTL_hg19_annot 

list.files(here::here('data/deCODE_pQTL/'),full.names = T) %>% 
  .[grep('.gz$',.)] %>% .[3] %>%
  as.list %>% 
  pblapply(format_pqtl_exposure,x.ref=ref.somalogic)
```

UKB pQTL

```{r}
format_pqtl_exposure <- function(f.x.dat,x.ref){
  # Get annotation
  Symbol = f.x.dat %>% basename %>% gsub('_adjAgeSexBatPC_InvNorm_22122022.txt.gz','',.) %>% gsub('Proteomics_SMP_PC0_','',.) 
  x.seqid = Symbol %>% str_split(.,'_') %>% .[[1]] %>% head(2) %>% paste0(.,collapse='_')
  x.ref = x.ref %>% filter(SeqId == x.seqid)
  x.chr = x.ref$CHR
  x.start = x.ref$start_pos
  x.end = x.ref$end_pos

  # Output name
  f.out = f.x.dat %>% basename %>% gsub('.txt.gz','_lo_annot.txt.gz',.)

  # Liftover from hg38 to hg19
  x.dat.liftover = read_tsv(f.x.dat) %>%
    dplyr::filter(Chrom!='chrX') %>%
    mutate(CHR=Chrom %>% gsub('chr','',.) %>% as.numeric,
           p=10^(-minus_log10_pval)) %>%
    dplyr::select(SNP=rsids,CHR,BP=Pos,MAF=ImpMAF,A1=effectAllele,A2=otherAllele,beta=Beta,se=SE,p=Pval,N,Name) %>% 
    MungeSumstats::liftover(sumstats_dt = ., 
                            ref_genome = "hg38",
                            convert_ref_genome = "hg19",end_col='BP') %>% 
    mutate(Phenotype=Symbol) 

  # Add variant annotation
  x.dat.liftover_annot = x.dat.liftover %>%
    dplyr::select(CHR,BP,SNP,effect_allele=A1,other_allele=A2,beta,se,pval=p,samplesize=N,Phenotype,Name) %>% 
    dplyr::filter(CHR==x.chr) %>% 
    left_join(.,annot_variant,by=c('Name')) %>%
    dplyr::select(-Name)
  
  # Write preprocessed file
  dir.create(here::here('./data/pQTL_hg19_annot/'), showWarnings = FALSE)
  write_tsv(x.dat.liftover_annot,here::here(paste0('data/pQTL_hg19_annot/',f.out)))
}

list.files(here::here('data/deCODE_pQTL/'),full.names = T) %>% 
  .[grep('.gz$',.)] %>% .[3] %>%
  as.list %>% 
  pblapply(format_pqtl_exposure,x.ref=ref.somalogic)
```


# Prepare BP sumstats

Download BP sumstats [here]( https://pgc.unc.edu/for-researchers/download-results/)

```{r, read gwas sumstats,eval=F,echo=T}
bp.gwas = fread('data/bip2024_eur_noUKB_no23andMe.gz')
```

```{r, rename sumstats colnames,eval=F,echo=T}
bp.gwas = bp.gwas %>% 
 mutate(beta = log(OR)) %>% 
 mutate(samplesize = Neff_half*2) %>%
 dplyr::select(SNP,CHR,BP,effect_allele=A1,other_allele=A2,INFO,
                       eaf=HRC_FRQ_A1,beta,se=SE,pval=P,samplesize) %>% 
         mutate(Phenotype='SCZ') 

write_tsv(bp.gwas,here::here('data/bp_2024.tsv.gz'))
```

A preprocessed copy is stored on cloud.