---
title: "MQ-DATAMIND ECR workshop: Proteomics and Mental Health (tutorial - local version)"
author: "X Shen"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  github_document
---

Welcome! This is the repository for the MQ-DATAMIND workshop, 'Proteomics and Mental Health' (Practical Session II)

# Aim of the session

We will use two-sample Mendelian randomisation to test causal relationship between protein abundance and schizophrenia. Through this tutorial, you will gain hands-on experience in handling and performing quality check on pQTL and other types of GWAS sumstats. You will also learn how to interpret Mendelian Randomisation findings.

# Data used in this tutorial and relevant readings

------------------------------------------------------------------------

We will use publicly available summary statistics from the [Psychiatric Genomic Consortium](https://pgc.unc.edu/for-researchers/download-results/) and pQTL data from [deCODE](https://www.decode.com/summarydata/) (Ferkingstad, E. et al. Large-scale integration of the plasma proteome with genetics and disease). Note that the complete pQTL data requires a simple access sign up. You can download your own version for your analysis following the link, or you could follow this tutorial using a local copy of pre-processed pQTL data.

Readings:

-   **pQTL**: Eldjarn, G.H., Ferkingstad, E., Lund, S.H. et al. Large-scale plasma proteomics comparisons through genetics and disease associations. Nature 622, 348‚Äì358 (2023). https://doi.org/10.1038/s41586-023-06563-x

-   **Schizophrenia GWAS**: Trubetskoy, V., Pardi√±as, A.F., Qi, T. et al. Mapping genomic loci implicates genes and synaptic biology in schizophrenia. Nature 604, 502‚Äì508 (2022). https://doi.org/10.1038/s41586-022-04434-5

-   **Mendelian Randomisation analysis**: Bhattacharyya U, John J, Lam M, Fisher J, Sun B, Baird D, Burgess S, Chen CY, Lencz T. Circulating Blood-Based Proteins in Psychopathology and Cognition: A Mendelian Randomization Study. JAMA Psychiatry. 2025 May 1;82(5):481-491. https://doi.org/10.1001/jamapsychiatry.2025.0033.

# Data preparation

------------------------------------------------------------------------

## Clone this github repo

Use the script below to clone this github repository on your local machine:

```{bash, eval=F, echo=T}
git clone https://github.com/xshen796/Proteomics_Workshop_Practical.git
```

## Set up your R environment to access OpenGWAS data

üí°See a quick guide here if you haven't done so: [URL](https://github.com/xshen796/Proteomics_Workshop_Practical/blob/main/Session_ii/Setup_APItoken.md)

## Data preparation

**See [here](https://github.com/xshen796/Proteomics_Workshop_Practical/blob/main/prep/data_prep.md) if you need more information on how we accessed and prepared the data for this tutorial.**

# Two-sample MR: finding causal proteins to schizophrenia using cis pQTL

------------------------------------------------------------------------

## Load R packages

```{r,warning=F,message=F}
library(rmarkdown)
library(dplyr)
library(data.table)
library(pbapply)
library(readr)
library(Hmisc)
library(here)
library(knitr)
library(kableExtra)
library(TwoSampleMR)
library(coloc)
```

## Load data

Let's first load schizophrenia GWAS sumstats and pQTL data

```{r load sumstats,message=F,warning=F,echo=T}
dat.scz_gwas = read_tsv('https://storage.googleapis.com/mhp-proteomic-sumstats/scz_2022.tsv.gz')

dat.pqtl_ptn = read_tsv('https://storage.googleapis.com/mhp-proteomic-sumstats/Proteomics_SMP_PC0_3045_72_PTN_PTN_10032022_lo_annot.txt.gz') %>% mutate(eaf = effectAlleleFreq)

dat.pqtl_lag3 = read_tsv('https://storage.googleapis.com/mhp-proteomic-sumstats/Proteomics_SMP_PC0_9950_229_LAG3_LAG_3_10032022_lo_annot.txt.gz')
```

Let's load protein annotation and see which proteins we will be looking at

```{r load annotation,message=F,echo=T}
ref.somalogic = read_tsv(here::here('utils/ref_somalogic.tsv')) %>%
    dplyr::select(SeqId,UniProt,Protein_full_name,Gene,Ensembl_Gene_ID)

knitr::kable(ref.somalogic, "pipe")
```

Here in this table:

-   SeqID: unique protein identifier of SOMALogic assay

-   UniProt, Protein_full_name: Uniprot ID and full name of the protein

-   Gene, Ensembl_Gene_ID: Gene symbol and Ensembl ID

## QC and prepare pQTL data (protein as exposure)

‚û°Ô∏è We need to select the cis region of ***PTN***. Find genomic locations for ***PTN*** [here](https://www.genecards.org/cgi-bin/carddisp.pl?gene=PTN&keywords=PTN)

Our genome build is **GRCh37/hg19**.

![PTN location](ptn_location.png)

```{r,eval=T,message=F,echo=T}
gene_boundary = data.frame(lower=136912088,upper=137028479)
gene_chr = 7
```

Select cis region sumstats: 1Mb extended area of the gene region

```{r,eval=T,message=F,echo=T}
dat.cis_pqtl_ptn = dat.pqtl_ptn %>% 
  dplyr::filter(CHR==gene_chr,BP > (gene_boundary$lower - 1000000), BP < (gene_boundary$upper + 1000000))
```

Double check if data only includes cis region

```{r,eval=T,message=F,echo=T}
range(dat.cis_pqtl_ptn$BP)
```

‚û°Ô∏è Keep common variants.

```{r,eval=T,message=F,echo=T}
dat.cis_pqtl_ptn = dat.cis_pqtl_ptn %>% filter(eaf < 0.995, eaf > 0.005)
```

‚û°Ô∏è Keep variants that also present in the schizophrenia GWAS sumstats.

```{r,message=F,warning=F}
dat.cis_pqtl_ptn = dat.cis_pqtl_ptn %>% filter(SNP %in% dat.scz_gwas$SNP)
```

‚û°Ô∏è Select genome-wide significant genetic instruments (pval \< 5e-8).

```{r,eval=T,message=F,echo=T}
dat.cis_sig_pqtl_ptn = dat.cis_pqtl_ptn %>% 
  dplyr::filter(pval<5e-8)
```

‚û°Ô∏è Clump data using the 1000 Genome central European genotype data as reference.

```{r,eval=T,message=F,echo=T}
dat.cis_sig_pqtl_ptn = dat.cis_pqtl_ptn %>% 
  clump_data(.,clump_kb=1000,clump_r2=0.001)
```

If you see any error message, it is likely that you have not set up your R environment correctly. See this [tutorial](https://github.com/xshen796/Proteomics_Workshop_Practical/blob/main/Session_ii/Setup_APItoken.md).

‚û°Ô∏è Reformat exposure data (pQTL)

```{r,warning=F,message=F}
dat.exposure.ptn = format_data(dat.cis_sig_pqtl_ptn,type='outcome')
```

## QC and prepare schizophrenia sumstats (schizophrenia as outcome)

```{r, format scz gwas,eval=T,message=F,echo=T}
dat.scz.outcome = dat.scz_gwas %>% 
  format_data(.,type='outcome') 
```

## Harmonise exposure and outcome data

Harmonise exposure and outcome data. Select the top SNP for Wald ratio test.

```{r, data harmonisation,eval=T,message=F,echo=T}
dat.mr.ptn_scz =  harmonise_data(x.exposure.ptn,dat.scz.outcome) %>%
    arrange(pval.exposure) %>%
    head(n=1)
```

## Run two-sample Mendelian randomisation (Wald ratio test and sign)

‚û°Ô∏è Use the script below to run Wald ratio test. See result below:

`{MR.wr=mr_wald_ratio(b_exp = dat.mr$beta.exposure,}                        b_out = dat.mr$beta.outcome,                        se_exp = dat.mr$se.exposure,                        se_out = dat.mr$se.outcome)  MR.wr %>% knitr::kable(.)`

Here, b = \`r MR.wr\$b %\>% round(.,digits=3), p = \`r MR.wr\$pval\`. This means a potentially positive causal association between plasma PTN abundance and liability of schizophrenia.

‚û°Ô∏è Use the script below to run MR Steiger directionality test. Here, we use it as a complementary analysis to Wald ratio.

```{r mr steiger, message=F,warning=F}
MR.dir = directionality_test(dat.mr)
MR.dir %>% %>% knitr::kable(.)
```

# Colocalisation: consolidating causal evidence

------------------------------------------------------------------------

We will use the cis region pQTL data generated above for MR analyses for colocalisation analysis. This time, we will use all the SNPs within the cis region.